### ISSUE TITLE
Wrong mouse coordinates after window resize when using viewport stretch-mode canvas_items or viewport

### ISSUE THREAD

**Issue by ZuBsPaCe** [godotengine/godot#52295]
### Godot version

v4.0.dev (75697c0df)

### System information

Windows 10

### Issue description

I want to position a Sprite2D at the global mouse position.

This works when setting stretch mode to disabled/keep-aspect, even when resizing the screen:
![ViewportScalingDisabled](https://user-images.githubusercontent.com/4233942/131581209-bf7ebde5-3d96-4b17-9bf2-88af8ac446ca.gif)

When I set stretch mode to canvas_items/keep-aspect, the mouse coordinates are sometimes wrong after I resize the window:
![ViewportScalingCanvasItems](https://user-images.githubusercontent.com/4233942/131581221-a4b392c6-0159-4fa4-b478-1d248e1187d4.gif)

When I set stretch mode to viewport/keep-aspect, the mouse coordinates are always wrong after I resize the window:
![ViewportScalingViewport](https://user-images.githubusercontent.com/4233942/131581231-984ad054-94b6-409f-a244-eb7656b6a7b4.gif)

Hints:
* I could not reproduce the Issue in 3.3.3-stable.
* Could be related to [changes regarding stretch mode](https://github.com/godotengine/godot-proposals/issues/2479#issuecomment-802998869).
* If I use an Area2D, mouse_entered and mouse_exited signals are also affected.
* Issue #34805 could be related. But it's in 3.2 and I'm not using ViewportContainer/Viewport at all.

### Steps to reproduce

1) Add a Sprite2D scene with following code:

```gdscript
func _process(delta):
    position = get_global_mouse_position()
```

2) In the project settings select viewport/keep-aspect or canvas_items/keep-aspect.
3) Run the game.
4) Resize the window by dragging the top, right, bottom or left window border.

### Minimal reproduction project

[ViewportScalingGodot4.zip](https://github.com/godotengine/godot/files/7087039/ViewportScalingGodot4.zip)

**Comment by CosmicSpore** [comment]
This bug does exist in v3.3 (and also v3.2 at the very least) but in a slightly different way...
Instead of global_mouse_position() not returning proper coordinates, the same problem instead occurs with **InputEvents**, particularly with "**event.global_position**". In fact, ironically enough, using global_mouse_position() actually seems to FIX the bug in v3.3. (So I guess technically this could be considered a different bug? But it seems similar enough to me that I figured I would just post here.)

Steps to reproduce on 3.3:
1. Add Sprite to scene with the following code:
```
func _input(event):
    if event is InputEventMouse:
	position = event.global_position
```
2. In project settings select: Mode: 2D, Stretch: Keep Height (May occur with other settings as well)
3. Run
4. Resize the window

Minimal project:
[TestTestTest.zip](https://github.com/godotengine/godot/files/7130461/TestTestTest.zip)

I've had a longstanding issue with this and I could never figure out why it happened until I saw this ticket. It never occurred to me the problem may have been in the engine itself. So I started testing a couple things and thanks to @ZuBsPaCe, for posting this bug, I finally found a way to work around it properly in my project. :)

**Comment by Racinettee** [comment]
I'm not fully sure if its the same issue or not, but I think this might also affect UI elements in godot 4.
When using the ItemList in godot 4, and the viewport mode: after expanding the window or using a larger test_mode size, the UI elements cannot be clicked anymore.
I couldn't reproduce the issue on 3.3 using the above settings, but in godot 4 the viewport setting on its own causes the UI elements to seem to not get mouse input.
